src/entry.rs

```diff
@@ -525,17 +527,29 @@ impl<'a> EntryFields<'a> {
 
         // Ensure we write a new file rather than overwriting in-place which
         // is attackable; if an existing file is found unlink it.
-        fn open(dst: &Path) -> io::Result<std::fs::File> {
-            OpenOptions::new().write(true).create_new(true).open(dst)
+        let mode = self.header.mode();
```

Comment:
Could this use `.ok()` to discard the error since it's ignored?

Reply:
that would be more awkward I think - because we only ignore the error to decide that readonly is false, so we'd be pulling apart the readonly logic  - we'd have to immediately evaluate the readonliness; whereas by keeping the monad we can use mode for the unix code paths as well.

Reply:
Er my point is that passing around `&io::Result<u32>` is just pretty weird, so it would be more idiomatic to pass around `Option<u32>`

Reply:
Oh, I get you now, thank you.

Reply:
Converted to Option<u32>